src/lib.rs

```diff
@@ -0,0 +1,1159 @@
+			self.channel_manager.as_ref(),
+		) {
+			Ok(_payment_id) => {
+				let payee_pubkey = invoice.recover_payee_pub_key();
+				let amt_msat = invoice.amount_milli_satoshis().unwrap();
+				log_info!(self.logger, "Initiated sending {} msats to {}", amt_msat, payee_pubkey);
+
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: payment_secret,
+						status: PaymentStatus::Pending,
+						amount_msat: invoice.amount_milli_satoshis(),
+					},
+				);
+
+				Ok(payment_hash)
+			}
+			Err(payment::PaymentError::Invoice(e)) => {
+				log_error!(self.logger, "Failed to send payment due to invalid invoice: {}", e);
+				Err(Error::InvalidInvoice)
+			}
+			Err(payment::PaymentError::Sending(e)) => {
+				log_error!(self.logger, "Failed to send payment: {:?}", e);
+
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: payment_secret,
+						status: PaymentStatus::Failed,
+						amount_msat: invoice.amount_milli_satoshis(),
+					},
+				);
+				Err(Error::PaymentFailed)
+			}
+		}
+	}
+
+	/// Send a spontaneous, aka. "keysend", payment
+	pub fn send_spontaneous_payment(
+		&self, amount_msat: u64, node_id: &str,
+	) -> Result<PaymentHash, Error> {
+		if self.running.read().unwrap().is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let mut outbound_payments_lock = self.outbound_payments.lock().unwrap();
+
+		let pubkey = hex_utils::to_compressed_pubkey(node_id).ok_or(Error::PeerInfoParseFailed)?;
+
+		let payment_preimage = PaymentPreimage(self.keys_manager.get_secure_random_bytes());
+		let payment_hash = PaymentHash(Sha256::hash(&payment_preimage.0).into_inner());
+
+		let route_params = RouteParameters {
+			payment_params: PaymentParameters::from_node_id(
+				pubkey,
+				self.config.default_cltv_expiry_delta,
+			),
+			final_value_msat: amount_msat,
+		};
+
+		match self.channel_manager.send_spontaneous_payment_with_retry(
+			Some(payment_preimage),
+			PaymentId(payment_hash.0),
+			route_params,
+			Retry::Timeout(LDK_PAYMENT_RETRY_TIMEOUT),
+		) {
+			Ok(_payment_id) => {
+				log_info!(self.logger, "Initiated sending {} msats to {}.", amount_msat, node_id);
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: None,
+						status: PaymentStatus::Pending,
+						amount_msat: Some(amount_msat),
+					},
+				);
+				Ok(payment_hash)
+			}
+			Err(e) => {
+				log_error!(self.logger, "Failed to send payment: {:?}", e);
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: None,
+						status: PaymentStatus::Failed,
+						amount_msat: Some(amount_msat),
+					},
+				);
+				Err(Error::PaymentFailed)
+			}
+		}
+	}
+
+	/// Returns a payable invoice that can be used to request and receive a payment.
+	pub fn receive_payment(
+		&self, amount_msat: Option<u64>, description: &str, expiry_secs: u32,
```

Comment:
Could you document `amount_msat`? Do we care that we don't have a separate method for zero-amount invoices?